tools/awslint/GUIDELINES.md

```diff
@@ -0,0 +1,295 @@
+# AWS Construct Library Design Guidelines
+
+The AWS Construct Library is a rich class library of CDK constructs which
+represent all resources offered by the AWS Cloud.
+
+The purpose of this document is to describe common guidelines for designing the
+APIs in the AWS Construct Library in order to ensure a consistent and integrated
+experience across the entire AWS surface area.
+
+As much as possible, the guidelines in this document are enforced using the
+__awslint__ tool which reflects on the APIs and verifies that the APIs adhere to
+the guidelines.
+
+When a guideline is backed by a linter rule, the rule name will be referenced
+like this: <a id="resource-class-is-construct">`awslint|resource-class-is-construct`</a>
+and anchored with the rule name.
+
+For the purpose of this document we will use "**Xxx**" to denote the official
```

Comment:
Maybe better to use a different magic word? `MyConstruct`?

Reply:
What's wrong with `Xxx`? If you are concerned with the connotation then maybe we can do `Foo` or something like that.

Reply:
Changing to "Foo"